hail/python/hail/expr/expressions/base_expression.py

```diff
@@ -388,10 +388,17 @@ def _promote_numeric(self, typ):
         coercer = expressions.coercer_from_dtype(typ)
         if isinstance(typ, tarray) and not isinstance(self.dtype, tarray):
             return coercer.ec.coerce(self)
+        elif isinstance(typ, tndarray) and not isinstance(self.dtype, tndarray):
+            return coercer.ec.coerce(self)
         else:
             return coercer.coerce(self)
 
-    def _bin_op_numeric_unify_types(self, name, other):
+    @staticmethod
```

Comment:
I think these changes need to be in a separate PR - they need a deeper review. Is that a lot of trouble?

Reply:
np I can move them

Reply:
Should be good now